wsinfer/write_geojson.py

```diff
@@ -64,62 +66,52 @@ def _dataframe_to_geojson(df: pd.DataFrame, prob_cols: list[str]) -> dict:
-@click.argument(
-    "output",
-    type=click.Path(exists=False, path_type=Path, resolve_path=True),
-)
-def togeojson(*, results_dir: Path, output: Path) -> None:
-    """Convert model outputs to GeoJSON format.
+def parallelize_geojson(csvs: list, results_dir: Path, num_workers: int) -> None:
+    output = results_dir / "model-outputs-geojson"
 
-    GeoJSON files can be used with pathology viewers like QuPath.
-
-    RESULTS_DIR     Path to results directory (containing model-outputs dir).
-
-    OUTPUT          Path to output directory in which to save GeoJSON files.
-    """
     if not results_dir.exists():
-        raise click.ClickException(f"results_dir does not exist: {results_dir}")
-    if output.exists():
-        raise click.ClickException("Output directory already exists.")
+        raise FileExistsError(f"results_dir does not exist: {results_dir}")
     if (
-        not (results_dir / "model-outputs").exists()
+        not (results_dir / "model-outputs-csv").exists()
         and (results_dir / "patches").exists()
     ):
-        raise click.ClickException(
+        raise FileExistsError(
             "Model outputs have not been generated yet. Please run model inference."
         )
-    if not (results_dir / "model-outputs").exists():
-        raise click.ClickException(
+    if not (results_dir / "model-outputs-csv").exists():
+        raise FileExistsError(
             "Expected results_dir to contain a 'model-outputs' directory but it does"
             " not. Please provide the path to the directory that contains"
             " model-outputs, masks, and patches."
         )
-
-    csvs = list((results_dir / "model-outputs").glob("*.csv"))
-    if not csvs:
-        raise click.ClickException("No CSVs found. Did you generate model outputs?")
-
-    output.mkdir(exist_ok=False)
-
-    for input_csv in tqdm.tqdm(csvs):
-        output_path = output / input_csv.with_suffix(".json").name
-        convert(input=input_csv, output=output_path)
-
-    click.secho(f"Saved outputs to {output}", fg="green")
+    if output.exists():
+        geojsons = list((results_dir / "model-outputs-geojson").glob("*.json"))
+        
+        # Makes a list of filenames for both geojsons and csvs
+        geojson_filenames = [filename.name.split(".")[0] for filename in geojsons]
+        csv_filenames = [filename.name.split(".")[0] for filename in csvs]
+
+        # Makes a list of new csvs that need to be converted to geojson
+        csvs_new = [filename for filename in csv_filenames if filename not in geojson_filenames]
+        csvs_final = [path for path in csvs if path.name.split(".")[0] in csvs_new]
+
+        # if len(csvs_final) == 0:
+        #     raise FileExistsError("All outputs have been converted to geojson")
+    else:
+        # If output directory doesn't exist, make one and set csvs_final to csvs
+        output.mkdir(parents=True, exist_ok=True)
+        csvs_final = csvs
+
+    func = partial(make_geojson, results_dir)
```

Comment:
use a keyword argument here instead of positional. it will make it clearer what is happening.

```python
func = partial(make_geojson, results_dir=results_dir)
```

Reply:
Yes, I tried using a keyword argument, but I got a TypeError error: `TypeError: make_geojson() got multiple values for argument 'results_dir'`, which is why I stuck to just passing it as a positional argument.

Reply:
ah! this is good to know. the order of the arguments in `make_geojson` needs to be flipped. the first argument should be the CSV, because that's the thing we are iterating over in the parallel mapper.

```python
def make_geojson(csv: Path, results_dir: Path) -> None:
    ...
```

in the previous version where `results_dir` was the first argument, the function was essentially being called as:

```python
make_geojson(csv=results_dir, results_dir=csv)
```

does that make sense?

Reply:
Ah yes, that does make sense. So I should be calling it like this, right? 
```python
    func = partial(make_geojson, results_dir = results_dir)
    process_map(func, csvs_final, max_workers=num_workers)
```

The whole point of partial is to fix certain arguments and we're doing that by fixing results_dir. Now that I think of it, we can just set it in the function definition itself. What do you think?

Reply:
the results_dir can vary and there isn't a reasonable default imho.

the issue is that `process_map` uses the first argument to the function. like this:

```python
for csv in csvs:
    func(csv)
```

that is equivalent to `process_map(func, csvs)`

but `make_geojson` has results_dir as the first argument. even if we fix `results_dir`, func will still set csv to the first argument, which in the current form is `results_dir`.